### PR TITLE
Essentials player balance event

### LINKED ISSUE #255: Essentials BalanceUpdateEvent
Basically a simple shortcut to get when a player's balance is changed until a better "more rounded method" as Morphan put it is found. API for event [here](https://github.com/EssentialsX/Essentials/blob/849efa9756495a34260d57ff1d81df1283e5f5e7/Essentials/src/net/ess3/api/events/UserBalanceUpdateEvent.java).

### PR DESCRIPTION
**Pull request by Ph4i1ur3**
Clean up Essentials bridge imports, implement new event for balance change.

Fixes #255

### CODE REVIEW

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import net.ess3.api.events.UserBalanceUpdateEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class EssentialsPlayerBalanceChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+
+    // <--[event]
+    // @Events
+    // player balance change
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: ^ typo (forgot the `s`)

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import com.denizenscript.denizencore.scripts.containers.ScriptContainer;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import net.ess3.api.events.UserBalanceUpdateEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class EssentialsPlayerBalanceChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: ^ don't have double-newlines... just one newline per spot

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    //
+    // @Cancellable false
+    //
+    // @Triggers when a player's balance changes, when using essentials economy.
+    //
+    // @Switch in <area>
+    //
+    // @Context
+    // <context.oldbalance> Returns the balance before changes are made
+    // <context.newbalance> Returns the balance after changes are made
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: put `_` between words, like `old_balance`

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    // <--[event]
+    // @Events
+    // player balance change
+    //
+    // @Regex ^on essentials player balance changes$
+    //
+    // @Cancellable false
+    //
+    // @Triggers when a player's balance changes, when using essentials economy.
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: capital E, `Essentials`, when you're using a proper noun in a sentence.

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+public class EssentialsPlayerBalanceChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+
+    // <--[event]
+    // @Events
+    // player balance change
+    //
+    // @Regex ^on essentials player balance changes$
+    //
+    // @Cancellable false
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: Just leave off the line entirely when it's false.

**Comment by Ph4i1ur3** [comment]
fixed

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import net.ess3.api.events.UserBalanceUpdateEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class EssentialsPlayerBalanceChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player balance changes
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: Missing the word `essentials`

File: src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    //
+    // @Regex ^on essentials player balance changes$
+    //
+    // @Triggers when a player's balance changes, when using Essentials economy.
+    //
+    // @Switch in <area>
+    //
+    // @Context
+    // <context.old_balance> Returns the balance before changes are made
+    // <context.new_balance> Returns the balance after changes are made
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/events/essentials/EssentialsPlayerBalanceChangeScriptEvent.java]: Periods at the end of the two context lines.

**Comment by Ph4i1ur3** [comment]
Corrections made

**Comment by mcmonkey4eva** [comment]
Yay!
Thanks!

**Pull request merged**